### PR TITLE
Fix location restriction

### PR DESCRIPTION
**Pull request by benrudolph**
@esoergel @czue this allows mobile works to see the cloudcare tab: http://manage.dimagi.com/default.asp?243701

it also does some work on viewing tabs for location restrictions. there's one bit i'm not sure the best way to fix, will mark in PR

cc: @emord

### CODE REVIEW

File: corehq/tabs/uitab.py [hunk 1]
```diff
@@ -126,9 +126,14 @@ def should_show(self):
...
-        if not self.can_access_all_locations and not self.filtered_dropdown_items:
-            # location-safe filtering makes this whole tab inaccessible
-            return False
+        if not self.can_access_all_locations:
+            if self.dropdown_items and not self.filtered_dropdown_items:
+                # location-safe filtering makes this whole tab inaccessible
+                return False
+
+            # Just a button tab, determine if it's location safe
+            if not self.dropdown_items and not url_is_location_safe(self.url):
```
Comment (emord) [inline corehq/tabs/uitab.py]: I think the dashboard should be visible to mobile workers https://github.com/dimagi/commcare-hq/pull/13268

Comment (benrudolph) [inline corehq/tabs/uitab.py]: Yes, but only if they have access to reports. I tested on a completely unprivileged mobile worker which gets redirected to the LocationList page and then renders a 403: https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/dashboard/views.py#L49

Comment (esoergel) [inline corehq/tabs/uitab.py]: The Orgs initiative is planning to tackle this at some point soon: https://trello.com/c/mIQiaisw/25-tb-ucr-making-dashboard-location-safe

**Review by esoergel — APPROVED** [review]

**Comment by czue** [comment]
thanks

**Pull request merged**